vpr/src/base/vpr_constraints.cpp

```diff
@@ -58,6 +60,70 @@ PartitionRegion VprConstraints::get_partition_pr(PartitionId part_id) {
     return pr;
 }
 
+void VprConstraints::add_route_constraint(RouteConstraint rc) {
```

Comment:
It looks to me that the current data structure is not efficient.
We have duplicated information which is the name of each routing constraint.
I suggest to merge and flatten the class ``RouteConstraint`` into ``VprConstraints``.
See an example at: 

https://github.com/lnis-uofu/OpenFPGA/blob/6adec23271b63ba112157b1f53afabf8e7bc7a20/libs/libbusgroup/src/bus_group.h#L108-L133